File: app/src/main/java/com/ahl/annahockeyleague/MainActivity.java
package com.ahl.annahockeyleague;


import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.fragment.NavHostFragment;
import androidx.navigation.ui.NavigationUI;

import android.os.Bundle;

import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        BottomNavigationView navView = findViewById(R.id.ahl_bottom_nav);

        NavHostFragment navHostFragment = (NavHostFragment) getSupportFragmentManager().findFragmentById(R.id.nav_host_fragment);
        NavigationUI.setupWithNavController(navView, navHostFragment.getNavController());
    }

}

File: app/src/main/java/com/ahl/annahockeyleague/MainActivity.kt
package com.ahl.annahockeyleague

import android.os.Bundle
import android.view.*
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.navigation.findNavController
import androidx.navigation.ui.setupWithNavController
import com.ahl.annahockeyleague.data.UIState
import com.ahl.annahockeyleague.ui.AhlViewModel
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity()  {

    private val ahlViewModel by viewModels<AhlViewModel>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        ahlViewModel.tournamentLiveData.observe(this, Observer { it ->
            when(it){

                UIState.SHOW_LOADER -> setErrorVisibility(errorVisibility = View.GONE)

                UIState.SHOW_CONTENT -> setErrorVisibility(errorVisibility = View.GONE)

                UIState.SHOW_ERROR -> setErrorVisibility(errorVisibility = View.VISIBLE)

                else -> return@Observer
            }
        })

        swipe_error.setOnClickListener {
            ahlViewModel.getAhlData()

        }

        ahlViewModel.getAhlData()
        ahl_bottom_nav.setupWithNavController(findNavController(R.id.nav_host_fragment))

    }



    private fun  setErrorVisibility(errorVisibility : Int){
        swipe_error.visibility = errorVisibility
    }

}